### PR TITLE
@W-11102262: grpc stream support

### PR DESCRIPTION
**Pull request by soaggarwal**

### CODE REVIEW

**Comment by g1-pai** [comment]
Please make sure all new dependencies added are apache licensed and no vulnerabilities.

File: src/main/java/com/salesforce/cdp/queryservice/util/Utils.java [hunk 1]
```diff
@@ -37,6 +38,15 @@ public class Utils {
             ExtendedHttpStatusCode.SC_BAD_REQUEST
     ));
 
+    private static final Set<Status.Code> grpcRetryStatusCodes = Collections.unmodifiableSet(Sets.newHashSet(
+            Status.Code.INTERNAL,
```
Comment (g1-pai) [inline src/main/java/com/salesforce/cdp/queryservice/util/Utils.java:42]: In what cases do we retry on internal errors

**Comment by g1-pai** [comment]
Please add unit tests

File: src/main/java/com/salesforce/cdp/queryservice/core/QueryServiceAbstractStatement.java [hunk 2]
```diff
@@ -122,6 +135,16 @@ private ResultSet createResultSetFromResponse(QueryServiceResponse queryServiceR
         return new QueryServiceResultSet(data, resultSetMetaData, this);
     }
 
+    private ResultSet createResultSetFromResponse(Iterator<AnsiSqlQueryStreamResponse> queryServiceResponse) throws SQLException {
+        if(queryServiceResponse.hasNext()) {
+            // first batch is metadata
```
[the remark this answers is not in the document]
Comment (soaggarwal) [reply src/main/java/com/salesforce/cdp/queryservice/core/QueryServiceAbstractStatement.java]: data is handled by the iterator in class QueryServiceHyperResultSet. line 142 here.

File: src/main/java/com/salesforce/cdp/queryservice/core/QueryServiceAbstractStatement.java [hunk 3]
```diff
@@ -122,6 +135,16 @@ private ResultSet createResultSetFromResponse(QueryServiceResponse queryServiceR
...
+            QueryServiceResultSetMetaData resultSetMetaData = createColumnNames(queryServiceResponse.next().getMetadata());
+            return new QueryServiceHyperResultSet(queryServiceResponse, resultSetMetaData, this);
+        }
+        // TODO: test this
+        throw new SQLException(queryServiceResponse.toString());
```
[the remark this answers is not in the document]
Comment (soaggarwal) [reply src/main/java/com/salesforce/cdp/queryservice/core/QueryServiceAbstractStatement.java]: if hasNext is false, that means even metadata is not present and that's an exception case always.

File: src/main/java/com/salesforce/cdp/queryservice/core/QueryServiceHyperResultSet.java [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+
+    private Object getValue(Object row, int columnIndex) throws SQLException {
+        return valueToObject(((ListValue) row).getValues(columnIndex));
+    }
+
+    private static Object valueToObject(Value value) {
+        switch (value.getKindCase()) {
+            case NULL_VALUE:
+                return null;
+            case NUMBER_VALUE:
```
[the remark this answers is not in the document]
Comment (soaggarwal) [reply src/main/java/com/salesforce/cdp/queryservice/core/QueryServiceHyperResultSet.java:103]: yes, checked

File: src/main/java/com/salesforce/a360/queryservice/grpc/v1/AnsiSqlExtractQueryRequest.java [hunk 5]
```diff
@@ -0,0 +1,1046 @@
+// Generated by the protocol buffer compiler.  DO NOT EDIT!
```
Comment (psasikaladevi) [inline src/main/java/com/salesforce/a360/queryservice/grpc/v1/AnsiSqlExtractQueryRequest.java:1]:
```suggestion
// Generated by the protocol buffer compiler.  DO NOT EDIT!
Lets not commit the extract changes now.
```

File: src/main/java/com/salesforce/a360/queryservice/grpc/v1/AnsiSqlExtractQueryRequestOrBuilder.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
+// Generated by the protocol buffer compiler.  DO NOT EDIT!
+// source: salesforce/cdp/queryservice/v1/query_service.proto
+
+package com.salesforce.a360.queryservice.grpc.v1;
+
+public interface AnsiSqlExtractQueryRequestOrBuilder extends
```
Comment (psasikaladevi) [inline src/main/java/com/salesforce/a360/queryservice/grpc/v1/AnsiSqlExtractQueryRequestOrBuilder.java:6]:
```suggestion
public interface AnsiSqlExtractQueryRequestOrBuilder extends
lets remove the extract related classes
```

File: src/main/java/com/salesforce/a360/queryservice/grpc/v1/AnsiSqlQueryRequest.java [hunk 7]
```diff
@@ -0,0 +1,1156 @@
+// Generated by the protocol buffer compiler.  DO NOT EDIT!
+// source: salesforce/cdp/queryservice/v1/query_service.proto
+
+package com.salesforce.a360.queryservice.grpc.v1;
+
+/**
+ * Protobuf type {@code salesforce.cdp.queryservice.v1.AnsiSqlQueryRequest}
+ */
+public final class AnsiSqlQueryRequest extends
```
Comment (psasikaladevi) [inline src/main/java/com/salesforce/a360/queryservice/grpc/v1/AnsiSqlQueryRequest.java:9]:
```suggestion
public final class AnsiSqlQueryRequest extends
We dont need this as well
```

**Review by psasikaladevi — APPROVED** [review]

**Pull request merged**